docs/standard/library-guidelines/strong-naming.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Strong Naming
+description: Best practice recommendations for strong-naming .NET libraries.
+author: jamesnk
+ms.author: jamesnk
+ms.date: 09/20/2018
+---
+# Strong Naming
+
+Strong naming refers to signing an assembly with a key, producing a strong named assembly. When an assembly is strong-named, it creates a unique identity based on the name and assembly version number, and can help prevent assembly conflicts.
+
+The downside to strong-naming is that the .NET Framework on Windows enables strict loading of assemblies once an assembly is strong named. A strong-named assembly reference must exactly match the version referenced by an assembly, forcing developers to [configure binding redirects](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/redirect-assembly-versions) when using the assembly:
+
+```xml
+<configuration>
+   <runtime>
+      <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+         <dependentAssembly>
+            <assemblyIdentity name="myAssembly" publicKeyToken="32ab4ba45e0a69a1" culture="neutral" />
+            <bindingRedirect oldVersion="1.0.0.0" newVersion="2.0.0.0"/>
+         </dependentAssembly>
+      </assemblyBinding>
+   </runtime>
+</configuration>
+```
+
+When .NET developers complain about strong-naming, what they're usually complaining about is strict assembly loading. Fortunately, this issue is isolated to the .NET Framework. .NET Core, Xamarin, UWP, and most other .NET implementations don't have strict assembly loading and removes the main downside of strong-naming.
+
+One important aspect of strong naming is it's viral: a strong named assembly can only reference other strong named assemblies. If your library isn't strong named then you have excluded developers who are building an application or library that needs strong naming from using it.
+
+The benefits of strong naming are:
+
+1. The assembly can be referenced and used by other strong-named assemblies.
+2. The assembly can be stored in the Global Assembly Cache (GAC).
+3. The assembly can be loaded side by side with other versions of the assembly. Side-by-side assembly loading is commonly required by applications with plug-in architectures.
+
+## What should I do?
+
+Applications don't need to strong name most of the time, but strong naming is recommended for .NET OSS libraries. Strong-naming an assembly ensures the most people can use it, and strict assembly loading only affects the .NET Framework.
+
+**✔️ CONSIDER** strong-naming your library's assemblies.
+
+**✔️ CONSIDER** checking in the key used to strong-name into your source control system.
```

Comment:
Public signing is only applicable if you absolutely cannot release the private key. Microsoft is in that camp because (1) our private keys are in hardware and (2) the same keys are used in older versions to enforce security rules. Virtually nobody is in that camp and I'd rather we don't complicate this guidance by mentioning it here.